### ISSUE TITLE
Registration doesn't work

### ISSUE THREAD

**Issue by kabukky** [go-authboss/authboss-sample#3]
Hey,
using `http://localhost:8080/auth/register` to register a new user doesn't seem to work. After the POST, there is no new user in the MemStorer, only the default one.
There is also no activity about a confirmation mail.
Since  `ab.Mailer = authboss.LogMailer(os.Stdout)` is set, I was assuming to see at least something about it in the console.
Is there something to configure first?

Thanks
Kai

**Comment by kris-runzer** [comment]
Hello!

Sorry for that; the issue is there was an error with the validation (it was looking for "username" instead of "email").

The default validators when you call authboss.New() are the following:

```
c.Policies = []Validator{
    Rules{
        FieldName:       "username",
        Required:        true,
        MinLength:       2,
        MaxLength:       4,
        AllowWhitespace: false,
    },
    Rules{
        FieldName:       "password",
        Required:        true,
        MinLength:       4,
        MaxLength:       8,
        AllowWhitespace: false,
    },
}
```

The default primaryID for authboss is "email', not "username".  I've have pushed the fix to authboss to prevent this issue:  [7cb7298](https://github.com/go-authboss/authboss/commit/7cb729827bb1db1f82f929e1c9eec0ddceb89e29)

Alternatively you can override the validators in blog.go

```
ab.Mailer = authboss.LogMailer(os.Stdout)

// Set new policies
ab.Policies = []authboss.Validator{
    authboss.Rules{
        FieldName:       "email",
        Required:        true,
        AllowWhitespace: false,
    },
    authboss.Rules{
        FieldName:       "password",
        Required:        true,
        MinLength:       4,
        MaxLength:       8,
        AllowWhitespace: false,
    },
}

if err := ab.Init(); err != nil {
    log.Fatal(err)
}
```

**Comment by kabukky** [comment]
Works like a charm now. Thanks :)
